components/dashboard/src/start/StartWorkspace.tsx

```diff
@@ -377,19 +377,37 @@ export default class StartWorkspace extends React.Component<StartWorkspaceProps,
         this.setState({ workspaceInstance, error });
     }
 
-    async ensureWorkspaceAuth(instanceID: string) {
+    async ensureWorkspaceAuth(instanceID: string, retry: boolean) {
         if (!document.cookie.includes(`${instanceID}_owner_`)) {
-            const authURL = gitpodHostUrl.asWorkspaceAuth(instanceID);
-            const response = await fetch(authURL.toString());
-            if (response.redirected) {
-                this.redirectTo(response.url);
-                return;
-            }
-            if (!response.ok) {
-                // getting workspace auth didn't work as planned - redirect
-                this.redirectTo(authURL.asWorkspaceAuth(instanceID, true).toString());
+            let attempt = 0;
+            while (attempt <= 10) {
+                attempt++;
+
+                const authURL = gitpodHostUrl.asWorkspaceAuth(instanceID);
+                const response = await fetch(authURL.toString());
```

Comment:
browser fetch may [throw errors](https://developer.mozilla.org/en-US/docs/Web/API/fetch#exceptions).

do we want to catch them or fails fast? specifically, the network errors would be nice to flatten out here.

Reply:
from scanning the call sites, it looks like it would be good to not throw.

Reply:
or at least catch in `onInstanceUpdate` handler.

Reply:
This fetch was there beforehand.

But I understand that you mean a lot of stuff might escape the try do to exceptions. So I will try to add it before merging. :+1:

Reply:
> This fetch was there beforehand.

Might be, but that seems to be broken, as `onInstanceUpdate` is called "manually", not just as event handler. 

The recent change doesn't fix it, I think. It remains unclear, how errors are handled in that area.

Reply:
> Might be, but that seems to be broken, as onInstanceUpdate is called "manually", not just as event handler.

Ah, this is it! Finally understood where the "intermittent redirect during a workspace session" was coming from: The StartWorkspace page is still present in the out frame, and still receives events :bulb: 
So that should be fixed by this.